src/pubtools/sign/signers/msgsigner.py

```diff
@@ -497,18 +505,24 @@ def msg_clear_sign_main(inputs, signing_key=None, task_id=None, config=None):
     type=str,
     help="References which should be signed.",
 )
+@click.option("--raw", default=False, is_flag=True, help="Print raw output instead of json")
 def msg_container_sign_main(
-    signing_key=None, task_id=None, config=None, digest=None, reference=None
+    signing_key=None, task_id=None, config=None, digest=None, reference=None, raw=None
 ):
     """Entry point method for containersign operation."""
-    click.echo(
-        json.dumps(
-            msg_container_sign(
-                signing_key=signing_key,
-                task_id=task_id,
-                config=config,
-                digest=digest,
-                reference=reference,
-            )
-        )
+    ret = msg_container_sign(
+        signing_key=signing_key,
+        task_id=task_id,
+        config=config,
+        digest=digest,
+        reference=reference,
     )
+    if not raw:
+        click.echo(json.dumps(ret))
+    else:
+        if ret["signer_result"]["status"] == "error":
+            print(ret["signer_result"]["error_message"], file=sys.stderr)
+            sys.exit(1)
```

Comment:
In case of failure, return code is 1 only if raw was specified? Is that intentional? Seems weird.